pkg/cmd/server/api/validation/node.go

```diff
@@ -27,9 +27,6 @@ func ValidateInClusterNodeConfig(config *api.NodeConfig, fldPath *field.Path) Va
 	if len(config.NodeName) == 0 {
 		validationResults.AddErrors(field.Required(fldPath.Child("nodeName"), ""))
 	}
-	if len(config.NodeIP) > 0 {
```

Comment:
restore validation, require it to be a valid IP, warn/error if it mismatches one specified in kubeletArgs[node-ip]